.github/workflows/haskell.yml

```diff
@@ -72,11 +72,13 @@ jobs:
         MSYSTEM: MINGW64
       run: cabal test all --enable-tests --test-show-details=direct -j1
 
-    - uses: actions/upload-artifact@v3
-      with:
-        name: Build & test logs
-        path: |
-          dist-newstyle/**/*.log
+    # v3 of this action has been deprecated and v4 errors out due to a name clash.
+    # Better to just disable it.
+    #- uses: actions/upload-artifact@v4
+    #  with:
+    #    name: Build & test logs
```

Comment:
I think you need `override: true` here. The semantics between v3 and v4 changed slightly. Iirc v3 did auto merge of artifacts under the same name. V4 doesn't.